python/ray/data/dataset.py

```diff
@@ -1836,6 +1846,8 @@ def iter_batches(self,
             A list of iterators over record batches.
         """
 
+        time_start = time.perf_counter()
```

Comment:
Could the entire code under a with statement: `self._stats.iter_total_s.timer()`?